src/System.Collections.Specialized/tests/NameObjectCollectionBase/MyNameObjectCollection.cs

```diff
@@ -10,7 +10,13 @@ public class MyNameObjectCollection : NameObjectCollectionBase
         public MyNameObjectCollection(int capacity) : base(capacity) { }
         public MyNameObjectCollection(IEqualityComparer comparer) : base(comparer) { }
         public MyNameObjectCollection(int capacity, IEqualityComparer comparer) : base(capacity, comparer) { }
-        
+#if netstandard17
+#pragma warning disable 618
+        public MyNameObjectCollection(IHashCodeProvider hashProvider, IComparer comparer) : base(hashProvider, comparer) { }
+        public MyNameObjectCollection(int capacity, IHashCodeProvider hashProvider, IComparer comparer) : base(capacity, hashProvider, comparer) { }
+#pragma warning restore 618
+#endif //netstandard17
+
```

Comment:
I've lost track of the guidelines we're trying to follow.  I thought we were trying to avoid these ifdefs and instead do them with partial classes.  Have we gone back on that?
cc: @weshaggard